### PR TITLE
27/Re-add support for publish.version

### PR DESCRIPTION
**Pull request by friedger**
**Problem**
Release 2.8 added support for project property "version". If this property was not set version "unspecified" was used instead of publish.version.

**Solution**
Check for "unspecified" when retrieving the property

**local maven for demo-project**
screen shot 2015-04-02 at 13 35 05

**local maven for bintray-release**
screen shot 2015-04-02 at 13 37 59

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ repositories {
 }
 
 dependencies {
-    compile 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.1'
+    compile 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.0'
```
Comment (malmstein) [inline core/build.gradle]: :100:

Comment (blundell) [inline core/build.gradle]: yea :+1: but two PR's please :x: or lets not open/merge this until #26 is merged

Comment (blundell) [inline core/build.gradle]: (or just revert this change for this PR)

**Comment by blundell** [comment]
@friedger you need to open a new PR - GitHub fails and still shows the diff pre the other merge sorry

**Comment by ouchadam** [comment]
I don't think this is the correct fix, we should avoid using `version` as a property

File: core/build.gradle (hunk 1 above)
Comment (xrigau) [inline core/build.gradle]: and also I'd say adding the demo project should be a separate PR as well to reduce the noise (fix vs. adding a demo project)

**Comment by friedger** [comment]
@ouchadam https://github.com/novoda/bintray-release/issues/22 says `version`as property name. As this is actually the version of the gradle project why not use it.

Some background for interested readers
http://stackoverflow.com/questions/13198358/how-to-get-project-version-in-custom-gradle-plugin/13198744#13198744

**Comment by ouchadam** [comment]
ah so version is the project.version still? it just needs to fetched after the project evaluation?

**Comment by ouchadam** [comment]
the is pr is still unreadable, you'll need to close and create a new one!

**Comment by friedger** [comment]
@ouchadam Can you clarify what is not readable? If I remove the standard generated Android project (that you have probably seen before) you will say that the PR is not testable. The demo project gives you the possibility to test the changes. I don't see any benefit in splitting the PR in two.

File: build.gradle [hunk 2]
```diff
@@ -1,3 +1,3 @@
 allprojects {
-    version = "0.2.8"
+    version = "0.2.9"
```
Comment (blundell) [inline build.gradle]: you should not up the version in this PR. Thats a separate concern that we can do afterwards :x:

File: demo-project/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    compileSdkVersion 22
+    buildToolsVersion "22.0.1"
+
+    defaultConfig {
+        minSdkVersion 15
+        targetSdkVersion 22
+        versionCode 1
+        versionName "1.0"
+    }
+    buildTypes {
```
Comment (blundell) [inline demo-project/app/build.gradle]: not needed for a demo

File: demo-project/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    groupId = 'com.example'
+    artifactId = 'myapplication'
+    version = '1.0.0'
+
+    description = 'An example library that uses the bintray-release plugin'
+    website = "https://github.com/novoda/bintray-release"
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (blundell) [inline demo-project/app/build.gradle]: not needed

File: demo-project/app/proguard-rules.pro [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# Add any project specific keep options here:
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
```
Comment (blundell) [inline demo-project/app/proguard-rules.pro]: not used

File: demo-project/app/src/androidTest/java/com/example/myapplication/ApplicationTest.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.example.myapplication;
+
+import android.app.Application;
+import android.test.ApplicationTestCase;
+
+/**
+ * <a href="http://d.android.com/tools/testing/testing_android.html">Testing Fundamentals</a>
+ */
+public class ApplicationTest extends ApplicationTestCase<Application> {
```
Comment (blundell) [inline demo-project/app/src/androidTest/java/com/example/myapplication/ApplicationTest.java]: not used

File: demo-project/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.example.myapplication">
+
+  <application android:allowBackup="true" android:label="@string/app_name" android:icon="@mipmap/ic_launcher" android:theme="@style/AppTheme">
```
Comment (blundell) [inline demo-project/app/src/main/AndroidManifest.xml]: formatting

File: demo-project/app/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<resources>
+
+  <!-- Base application theme. -->
+  <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <!-- Customize your theme here. -->
```
Comment (blundell) [inline demo-project/app/src/main/res/values/styles.xml]: usnued so why have it?

**Comment by blundell** [comment]
why not do the fix and add a demo in two separate PR's?

**Comment by malmstein** [comment]
Agree on not upping the version in this PR and have a separate one for bumping the version.
